src/com/sittinglittleduck/DirBuster/Worker.java

```diff
@@ -573,20 +321,176 @@ else if(work.getBaseCaseObj().isUseRegexInstead())
             }
             finally
             {
-                if(httpget != null)
-                {
-                    httpget.releaseConnection();
-                }
+            	if (httpMethod != null){
+            		httpMethod.releaseConnection();
+            	}
+            }
+        }
 
-                if(httphead != null)
-                {
-                    httphead.releaseConnection();
-                }
+    }
+    
+    private HttpMethodBase getHttpMethod (String method, String url) {
+    	return method.equalsIgnoreCase("HEAD") ? new HeadMethod(url) : new GetMethod(url);
+    }
+     
+    private int makeRequest(HttpMethodBase httpMethod) throws HttpException, IOException, InterruptedException {
+    	if(Config.debug)
+        {
+            System.out.println("DEBUG Worker[" + threadId + "]: "+ httpMethod.getName() + " : " + url.toString());
+        }
+
+        //set the custom HTTP headers
+        Vector HTTPheaders = manager.getHTTPHeaders();
+        for(int a = 0; a < HTTPheaders.size(); a++)
+        {
+            HTTPHeader httpHeader = (HTTPHeader) HTTPheaders.elementAt(a);
+            /*
+             * Host header has to be set in a different way!
+             */
+            if(httpHeader.getHeader().startsWith("Host"))
+            {
+            	httpMethod.getParams().setVirtualHost(httpHeader.getValue());
+            }
+            else
+            {
+            	httpMethod.setRequestHeader(httpHeader.getHeader(), httpHeader.getValue());
             }
+
         }
+        httpMethod.setFollowRedirects(Config.followRedirects);
 
+        /*
+         * this code is used to limit the number of request/sec
+         */
+        if(manager.isLimitRequests())
+        {
+            while(manager.getTotalDone() / ((System.currentTimeMillis() - manager.getTimestarted()) / 1000.0) > manager.getLimitRequestsTo())
+            {
+                Thread.sleep(100);
+            }
+        }
+        /*
+         * Send the head request
+         */
+        return httpclient.executeMethod(httpMethod);
     }
 
+	private boolean verifyIfCodeIsValid(int code) {
+		return code != HttpStatus.SC_NOT_FOUND && code != 0 && code != HttpStatus.SC_BAD_GATEWAY;
+	}
+
+	private void verifyResponseForOtherRequests(int code, String responce, String rawResponce) {
```

Comment:
Doesn't seem to be the best name, specially since it's doing the same as ```sendResponseToManager```.
I would replace this method with a call to ```sendResponseToManager```.